sdk/iot/hub/inc/az_iot_hub_client.h

```diff
@@ -89,14 +89,17 @@ AZ_NODISCARD az_result az_iot_hub_client_init(
  * {iothubhostname}/{device_id}/{module_id}/?api-version=2018-06-30&{user_agent}
  *
  * @param[in] client The #az_iot_hub_client to use for this call.
- * @param[in] mqtt_user_name An empty #az_span with sufficient capacity to hold the MQTT user name.
- * @param[out] out_mqtt_user_name The output #az_span containing the MQTT user name.
+ * @param[out] mqtt_user_name A char buffer with sufficient capacity to hold the MQTT user name.
+ * @param[in] mqtt_user_name_size The size of the passed buffer. Must be greater than 0.
+ * @param[out] out_mqtt_user_name_length The optional output length of the mqtt user name. Can
+ * be `NULL`.
  * @return #az_result.
  */
 AZ_NODISCARD az_result az_iot_hub_client_user_name_get(
     az_iot_hub_client const* client,
-    az_span mqtt_user_name,
-    az_span* out_mqtt_user_name);
+    char* mqtt_user_name,
+    int32_t mqtt_user_name_size,

```

Comment:
These must be `size_t`.

Reply:
Yea I think I ran into some issue using `size_t` when doing the pnp changes when casting. I can try it again